src/ComputeSharp.D2D1/Interfaces/__Internals/ID2D1InputDescriptionsLoader.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.ComponentModel;
+
+namespace ComputeSharp.D2D1.__Internals;
+
+/// <summary>
+/// A base <see langword="interface"/> representing a loader for input descriptions in a D2D1 shader.
+/// </summary>
+[EditorBrowsable(EditorBrowsableState.Never)]
+[Obsolete("This interface is not intended to be used directly by user code")]
+public interface ID2D1InputDescriptionsLoader
+{
+    /// <summary>
+    /// Loads the available input descriptions for the shader.
+    /// </summary>
+    /// <param name="data">The sequence of serialized input descriptions.</param>
+    /// <exception cref="InvalidOperationException">Thrown if the input descriptions have already been set.</exception>
```

Comment:
"if the input descriptions have already been set" ? Why would this method throw? This description doesn't make sense to me, since this method is for loading, not setting

Reply:
This is an internal interface (or rather, an implementation detail), but yeah this exception shouldn't be documented here. In fact, the actual loader implementation wasn't throwing either. I've fixed it in the latest force push 🙂

Reply:
Yeah future you or some future maintainer would be puzzled by this and worry that the code changed without updated docs, or that there's a bug!